kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/filter/DenyCipherSuiteFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import io.netty.handler.ssl.CipherSuiteFilter;
+import io.netty.util.internal.EmptyArrays;
+
+/**
+ * This class has been built to filter cipher suites based on requirements for Kroxylicious
+ * If no ciphers are declared then the default platform ciphers will be used
+ * If ciphers are declared then they are used instead of the default platform ciphers
+ * Both lists would then have anything removed that wasn't supported or in the deniedCiphers
+ */
+public final class DenyCipherSuiteFilter implements CipherSuiteFilter {
+
+    public static final DenyCipherSuiteFilter INSTANCE = new DenyCipherSuiteFilter();
+
+    private DenyCipherSuiteFilter() {
+    }
+
+    private List<String> deniedCiphers;
```

Comment:
Should be a `Set` I think

Reply:
`deniedCiphers` is a set.  The order of the denies doesn't matter.

`allowedCiphers` has to be ordered, as you say.     The best choice would be a `SequencedSet`, but we are not on Java 21 and SequencedSet wont play nicely with direction #1639 is going.   Modelling it as a List<String> will be fine. Your DenyCipherSuiteFilters#ciphers method impl will receive the allowedCiphers asthe  Iterable ciphers arg. Order will be maintained.

Reply:
Ah yes of course, denied doesn't matter, apologies @robobario you were right